### ISSUE TITLE
Using parsers for transforming

### ISSUE THREAD

**Issue by grigandal625** [aroberge/ideas#28]
It would be very useful to implement the search and processing of fragments of transformable code using parsers

For example, using [lark-parser](https://github.com/lark-parser/lark), we could make a transformation like this:

```python
# specifying a fragment grammar with lark syntax (EBNF-like)
# such a grammar could be intended for a fragment like:
# 5 $ 10
grammar = """
    new_operation: NUMBER '$' NUMBER 
    %import common.SIGNED_NUMBER -> NUMBER
"""

@detect(grammar)
def transform(parsed_tree, detected_source, full_source, **kwargs):
    # parsed_tree will be like:
    #
    # new_operation:
    # .... 5
    # .... 10

    return do_smth_with_tree(parsed_tree)
```

**Comment by aroberge** [comment]
There is absolutely nothing that prevents you from doing this.

Some existing examples require the use of other libraries. For example, [pythonji](https://github.com/aroberge/ideas/blob/master/ideas/examples/pythonji.py) requires the use of `emoji`. [easy_units](https://github.com/aroberge/ideas/blob/master/ideas/examples/easy_units.py) require either `pint` or `astropy`.

**Comment by grigandal625** [comment]
You are absolutely right, and I plan to do so. I just wanted to say that in solutions related to language processing, an approach using parsers can greatly simplify the processing task. Accordingly, in my opinion, it would be most appropriate to include such functionality in such solutions. Naturally, everything is at your discretion. Thanks

**Comment by aroberge** [comment]
If you do so, would you mind adding a link to this issue so that I could see if I could include a similar example in this project?

**Comment by grigandal625** [comment]
Shure

**Comment by grigandal625** [comment]
here is an [example](https://github.com/grigandal625/pyat-krl)
but just note that I haven't been careful with the architecture of the project yet, so direct copying can cause problems)

**Comment by aroberge** [comment]
Thanks.
